util/misc.go

```diff
@@ -559,13 +559,12 @@ func IsTLSExpiredError(err error) bool {
 		if inval.Reason != x509.Expired {
 			return false
 		}
-		// TODO: remove it after upgrading 1.20
-		// case *tls.CertificateVerificationError:
-		// 	invali, ok := inval.Err.(x509.CertificateInvalidError)
-		// 	if !ok || invali.Reason != x509.Expired {
-		// 		return false
-		// 	}
-		// 	return true
+	case *tls.CertificateVerificationError:
```

Comment:
It is related with go1.20. 

For a handshake failure due to a certificate verification failure, the TLS client and server now return an error of the new type [CertificateVerificationError](https://tip.golang.org/pkg/crypto/tls/#CertificateVerificationError), which includes the presented certificates.